apis/capabilities/v1beta1/developeruser_types.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2020 Red Hat.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+	"github.com/3scale/3scale-operator/pkg/helper"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	DeveloperUserKind = "DeveloperUser"
+
+	// DeveloperUserInvalidConditionType represents that the combination of configuration
+	// in the spec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	DeveloperUserInvalidConditionType common.ConditionType = "Invalid"
+
+	// DeveloperUserOrphanConditionType represents that the configuration in the spec
+	// contains reference to non existing resource.
+	// This is (should be) a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	// Example: the DeveloperUserSpec references non existing product resource
+	DeveloperUserOrphanConditionType common.ConditionType = "Orphan"
+
+	// DeveloperUserReadyConditionType indicates the activedoc has been successfully synchronized.
+	// Steady state
+	DeveloperUserReadyConditionType common.ConditionType = "Ready"
+
+	// DeveloperUserFailedConditionType indicates that an error occurred during synchronization.
+	// The operator will retry.
+	DeveloperUserFailedConditionType common.ConditionType = "Failed"
+
+	// DeveloperUserPasswordSecretField indicates the secret field name with developer user's password
+	DeveloperUserPasswordSecretField = "password"
+)
+
+// DeveloperUserSpec defines the desired state of DeveloperUser
+type DeveloperUserSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Username
+	Username string `json:"username"`
+
+	// Email
+	Email string `json:"email"`
+
+	// Password
+	PasswordCredentialsRef corev1.SecretReference `json:"passwordCredentialsRef"`
+
+	// DeveloperAccountRef is the reference to the parent developer account
```

Comment:
This means we only support specifying developer accounts that have been created through the capabilities functionality right?

Reply:
Right. 

Would be more flexible to make reference to Account ID? I think we should be generic and make references to other CR names or unique fields in the CR (like system_name in other CRD's). But I do not have strong opinion.

Reply:
I don't have a strong opinion about it either.